oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -992,13 +992,19 @@ public void getRequestMetadata(
     // For default universe Self-signed JWT could be explicitly disabled with
     // {@code ServiceAccountCredentials.useJwtAccessWithScope} flag.
     // If universe is non-default, it only supports self-signed JWT, and it is always allowed.
-    if (this.useJwtAccessWithScope || !isDefaultUniverseDomain()) {
-      // This will call getRequestMetadata(URI uri), which handles self-signed JWT logic.
-      // Self-signed JWT doesn't use network, so here we do a blocking call to improve
-      // efficiency. executor will be ignored since it is intended for async operation.
-      blockingGetToCallback(uri, callback);
-    } else {
-      super.getRequestMetadata(uri, executor, callback);
+    try {
+      if (this.useJwtAccessWithScope || !isDefaultUniverseDomain()) {
+        // This will call getRequestMetadata(URI uri), which handles self-signed JWT logic.
+        // Self-signed JWT doesn't use network, so here we do a blocking call to improve
+        // efficiency. executor will be ignored since it is intended for async operation.
+        blockingGetToCallback(uri, callback);
+      } else {
+        super.getRequestMetadata(uri, executor, callback);
+      }
+    } catch (IOException e) {
+      // Wrap it here to avoid breaking change.
```

Comment:
What would be a breaking change?

Reply:
added to comment, to throw IOException would be breaking.